### PR TITLE
Add config to disable dismissal of collapsed drawer

### PR DESCRIPTION
**Pull request by georgmay**
Hello!

Here's a PR to allow users disable dismissal of a drawer view and keep some height of it always on screen. 

I think i can reference [this](https://github.com/Babylonpartners/DrawerKit/issues/53) issue here.

### CODE REVIEW

**Comment by daxfrost** [comment]
I really want this too!

File: DrawerKitDemo/DrawerKitDemo/PresenterViewController.swift [hunk 1]
```diff
@@ -37,6 +37,7 @@ private extension PresenterViewController {
         configuration.upperMarkGap = 100 // default is 40
         configuration.lowerMarkGap =  80 // default is 40
         configuration.maximumCornerRadius = 15
+        configuration.neverDismissCollapsedDrawer = true
```
Comment (wltrup) [inline DrawerKitDemo/DrawerKitDemo/PresenterViewController.swift]: If changing from `neverDismissCollapsedDrawer` to `canDismissCollapsedDrawer`, then the value here should be `false`.

File: DrawerKit/DrawerKit/Internal API/PresentationController+Animation.swift [hunk 2]
```diff
@@ -71,7 +71,9 @@ extension PresentationController {
                     (isEndingStateCollapsed && endingPosition == .end)
 
             if shouldDismiss {
-                self.presentedViewController.dismiss(animated: true)
+                if !self.configuration.neverDismissCollapsedDrawer {
+                    self.presentedViewController.dismiss(animated: true)
+                }
```
[the remark this answers is not in the document]
Comment (georgmay) [reply DrawerKit/DrawerKit/Internal API/PresentationController+Animation.swift]: Makes sense, I'll fix it.

File: DrawerKit/DrawerKit/Public API/Protocols/DrawerPresentable.swift [hunk 3]
```diff
@@ -6,4 +6,8 @@ public protocol DrawerPresentable: class {
     /// The height at which the drawer must be presented when it's in its
     /// partially expanded state. If negative, its value is clamped to zero.
     var heightOfPartiallyExpandedDrawer: CGFloat { get }
+    
+    /// The height at which the drawer should be presented when it's in its
+    /// collapsed expanded state. If negative, its value is clamped to zero.
```
[the remark this answers is not in the document]
Comment (georgmay) [reply DrawerKit/DrawerKit/Public API/Protocols/DrawerPresentable.swift]: 👍

File: DrawerKit/DrawerKit/Internal API/PresentationController+Gestures.swift [hunk 4]
```diff
@@ -15,7 +15,10 @@ extension PresentationController {
         guard tapY < currentDrawerY else { return }
         NotificationCenter.default.post(notification: DrawerNotification.drawerExteriorTapped)
         tapGesture.isEnabled = false
-        presentedViewController.dismiss(animated: true)
+        
+        if !configuration.neverDismissCollapsedDrawer {
```
[the remark this answers is not in the document]
Comment (georgmay) [reply DrawerKit/DrawerKit/Internal API/PresentationController+Gestures.swift]: 👍

File: DrawerKit/DrawerKit/Internal API/GeometryEvaluator.swift [hunk 5]
```diff
@@ -89,11 +94,12 @@ extension GeometryEvaluator {
 
     static func drawerPositionY(for state: DrawerState,
                                 drawerPartialHeight: CGFloat,
+                                drawerCollapsedHeight: CGFloat,
                                 containerViewHeight: CGFloat,
                                 drawerFullY: CGFloat) -> CGFloat {
         switch state {
         case .collapsed:
-            return containerViewHeight
+            return containerViewHeight - drawerCollapsedHeight
```
Comment (wltrup) [inline DrawerKit/DrawerKit/Internal API/GeometryEvaluator.swift]: If I understand correctly, I think the subtraction here should only happen when `canDismissCollapsedDrawer` is `false`. If it's `true`, then the previous behaviour (ie, no subtraction) should be the one that takes effect. In other words, I think this should be

```swift
let offset: CGFloat = canDismissCollapsedDrawer ? 0 : drawerCollapsedHeight
return containerViewHeight - offset
```

Please make sure to test the behaviour for *both* values of `canDismissCollapsedDrawer` and make sure that the behaviour for `canDismissCollapsedDrawer == true` matches the current behaviour.

File: DrawerKit/DrawerKit/Public API/Structs/DrawerConfiguration.swift [hunk 6]
```diff
@@ -82,6 +82,9 @@ public struct DrawerConfiguration {
     /// When `false`, the presentation is always to full screen and there is no
     /// partially expanded state. The default value is `true`.
     public var supportsPartialExpansion: Bool
+    
+    /// When `false`, collapsed drawer is never dismissed.
```
Comment (wltrup) [inline DrawerKit/DrawerKit/Public API/Structs/DrawerConfiguration.swift]: This needs a comment about the default value (which is `true`). Something like:

/// When `false`, collapsed drawer is never dismissed. The default value is `true`.

**Comment by ilyapuchka** [comment]
I believe this is addressed now with #75 though without a configuration to not dismiss collapsed drawer

**Comment by wltrup** [comment]
I'm going to close this PR since the changes haven't been addressed, there are conflicts, and - as @ilyapuchka has pointed out - this is fixed by another PR.

**Pull request closed without merging**